### ISSUE TITLE
Allow manual tweaking of delta generation

### ISSUE THREAD

**Issue by alexlarsson** [ostreedev/ostree#314]
In flatpak, we typically have two related runtimes, the platform and the corresponding sdk. These share a lot of files, but some are different, and many are only in the sdk.

If using static deltas to download these two it seems like we're downloading everything in the platform twice. It would be nice if we could somehow make the sdk deltas refer to the platform ones, or otherwise set things up so that the sdk deltas have shared files exclusively in some delta parts so they can be completely omitted from the download.

**Comment by cgwalters** [comment]
Have you considered having the SDK depend on the runtime?  It's possible to implement ostree-native dependencies in a few ways, such as having a metadata key in the SDK commit which points to the runtime commit.

**Comment by alexlarsson** [comment]
What would such dependencies achieve though? Even if you have the platform downloaded, when you pull the deltas its quite likely that there is one or two files in each delta part, so we'll have to download them all, no?

**Comment by alexlarsson** [comment]
And, isn't that true even if you pulled the two branches in the same operation?

**Comment by cgwalters** [comment]
In this the SDK commit would not contain the runtime parts, and whenever creating a checkout of the SDK, flatpak would first check out the runtime tree.

**Comment by alexlarsson** [comment]
So, a Union checkout? That makes it impossible to have files in the platform that are not in the SDK. Maybe that's not a problem though.

**Comment by cgwalters** [comment]
I can't think of a case where one would want runtime-only content, and if it did exist it'd probably be small things that one wouldn't care about in terms of size, and could be disabled/overwritten by the SDK.

(For example, if one wanted a different `/usr/bin/foo` for some reason, union checkouts solve that by replacing)

**Comment by alexlarsson** [comment]
So, what i meant by tweaking the delta btw is that the delta format allows you to refer to other deltas. We can easily tell it to first include the platform deltas and then only package the rest in the sdk deltas.

**Comment by cgwalters** [comment]
Yeah, I understand.  It's possible to auto-detect the content duplication, but most of this is blocked on smarter delta generation tooling.  Related: https://bugzilla.gnome.org/show_bug.cgi?id=764009
